.github/workflows/build_bazel.yml

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 The StableHLO Authors.
+
+name: Bazel Build
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths: ['**.txt', 'WORKSPACE']
```

Comment:
This will only run a bazel build when we update a txt or workspace file. Is that the intention? Or do we want to run this for every job? Can omit this line all together if so.

Same for push.

Reply:
I think we should run for every job, unless this leads to significant slowdowns of our CI. 

How much additional CI time does the Bazel build take? If I understand correctly, it runs in parallel, so probably none or close to none?

Reply:
When cached, it's basically nothing. During llvm integrates, it runs for about an hour, but less than llvm build time, so it does not add additional time from what we have now.